### DISCUSSION TITLE
Running Nx commands from project apps directory not working

### DISCUSSION THREAD

**Discussion by lewsmith** [nrwl/nx#17416] [General]
Not sure when this started, but on my previous NX projects I could run any NX project commands in any sub dir and the command would work fine. However with the latest version of NX it appears that all commands fail to run unless I am running from the root of the nx project (containing the nx.json). 

Anyone else see this issue?

Lets say  I have a simple workspace, with one app that has some sort of run command:

```
my-project
├── apps
│   └── my-app
│       └── project.json
└── nx.json
```

If I run `nx run my-app:run` from the `my-project` directory all is good:
```
>  NX   Running target run for project my-app
```

If I run the same command, from within the `my-project/apps/my-app`, which definitly used to work:
```
>  NX   Ran target run for project my-app (685ms)
 
    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
```

I get no extra output if I run with the `--verbose` flag:
```
>  NX   Running target run for project my-app failed

   Failed tasks:
   
   - my-app:run
   
   Hint: run the command with --verbose for more details. <-- I get this message even thought I'm running with verbose (?)
```

Any clues?

```
   Node   : 18.16.0
   OS     : linux x64
   pnpm   : 8.3.1
   Hasher : Native
   
   nx (global)        : 16.2.1
   nx                 : 16.1.4
   @nx/js             : 16.1.4
   @nx/jest           : 16.1.4
   @nx/linter         : 16.1.4
   @nx/devkit         : 16.1.4
   @nx/eslint-plugin  : 16.1.4
   nx-cloud           : 16.0.5
   typescript         : 4.9.5
```

**Comment by ArminArdekani** [comment]
I'm having the same problem, I think `project.json` expects everything to run from the root level otherwise it starts creating build directories in places its not suppose if we run `nx run` within a project. I agree, this use to work fine before so perhaps a new bug.

**Comment by jimmcslim** [comment]
I agree that the ability to `nx run` from any directory within the project and have it "just work" would be amazing... I'm surprised that it doesn't?

**Comment by jimmcslim** [comment]
Scratching the itch on this one further... I have confirmed that this used to work in some way, although perhaps it was not by design.

I have one monorepo that is running 16.7.4 and if I run a target from deep within the folder structure it completes successfully, as it does if I run it from the root. I have a separate monorepo that is running 19.5.0 and attempting to run a target from anywhere other than the monorepo root fails.
